### PR TITLE
Fix monikers of toc node

### PR DESCRIPTION
**Pull request by 928PJY**
- If the node is a text node(`## Header`), the *monikers* of this node is the collection of its children's monikers.
- If the node is an absolute link node(`## [Header](/a)` or `## [Header](http://test)`), the *monikers* of this node is the monikers of this toc file, which means this node will always be displayed.

### CODE REVIEW

File: src/docfx/build/toc/BuildTableOfContents.cs [hunk 1]
```diff
@@ -82,11 +82,19 @@ void ResolveItemMonikers(List<TableOfContentsItem> items)
                     }
 
                     List<string> monikers = null;
+
+                    var linkType = UrlUtility.GetLinkType(item.Href?.Value);
+                    if (linkType == LinkType.External || linkType == LinkType.AbsolutePath)
```
Comment (yufeih) [inline src/docfx/build/toc/BuildTableOfContents.cs]: Instead of checking for link type, is it easier to detect pure text node?

Comment (928PJY) [reply src/docfx/build/toc/BuildTableOfContents.cs]: there are three cases need to check

Comment (mingwli) [reply src/docfx/build/toc/BuildTableOfContents.cs]: can we reuse [TocHrefType](https://github.com/dotnet/docfx/blob/b67e013094b9a7b868f60d86133a7e27943cd3c7/src/docfx/build/toc/parser/TableOfContentsParser.cs#L350) here?

Comment (928PJY) [reply src/docfx/build/toc/BuildTableOfContents.cs]: that require saving that info in the tocModel

**Pull request merged**